include/rosdiscover-clang/Callback/Callback.h

```diff
@@ -21,13 +21,22 @@ class Callback {
     llvm::outs() << "\n";
 
     auto *unaryOp = clang::dyn_cast<clang::UnaryOperator>(argExpr);
+    clang::Expr const *subExpr;
     if (unaryOp == nullptr) {
-      return unableToResolve(argExpr);
-    }
-
-    auto *subExpr = unaryOp->getSubExpr();
-    if (subExpr == nullptr) {
-      return unableToResolve(argExpr);
+      auto *castExpr = clang::dyn_cast<clang::ImplicitCastExpr>(argExpr);
```

Comment:
I'll give more detailed feedback on this after my meetings, but the code on the left is intentionally structured to reduce complexity. Rather than having nested if/else statements, you can have a flat chain of if statements where you `return unableToResolve(argExpr);` if something is wrong.